src/expint.jl

```diff
@@ -0,0 +1,28 @@
+function expint(x::Float64)
+    if x >= 1.0
+        e = exp(-x/2)
+        xinv = inv(x)
+        a = evalpoly(xinv, (1.3319248966682697359e-30, 0.99999999999999993242, 37.702667724670064050, 542.50695317051747114, 3836.6700640600929730, 14346.114698914059563, 28502.272377756659505, 28835.636437354567322, 13473.368739266960527,2387.9894006598233823, 99.579680699439479278))
+        b = evalpoly(xinv, (1.0, 38.702667724669835916, 579.20962089531281339, 4344.4743494790396875, 17740.385816044920402, 40220.102972229127618, 49665.114336529855022, 31552.009191297117961, 9222.0904372933369941, 992.44893809992759951,22.340799371096865866))
+        return (a / b * e) * e
+    else
+        a = evalpoly(x, (1.0000000000000000001,0.16967259504587971303,0.027806394103770057485,0.0014502330283706966415,0.000067149786258297615728,-5.9741062104757485179e-8))
+        b = evalpoly(x, (1,0.41967259504587972849,0.077168987309683782070,0.0078440023532535002066,0.00044590727789287393878,0.000011509944564265827774))
+        p = x * a / b
+        return (-(Base.MathConstants.eulergamma + log(x)) + p)
+    end
+end
```

Comment:
```suggestion
# power series minimax
const EXPINT_POW_COEF = (
    (1.0000000000000000001, 0.16967259504587971303, 0.027806394103770057485, 0.0014502330283706966415, 0.000067149786258297615728, -5.9741062104757485179e-8),
    (1.0, 0.41967259504587972849, 0.077168987309683782070, 0.0078440023532535002066, 0.00044590727789287393878, 0.000011509944564265827774),
)

# asymptotic expansion minimax
const EXPINT_ASYM_COEF = (
    (1.3319248966682697359e-30, 0.99999999999999993242, 37.702667724670064050, 542.50695317051747114, 3836.6700640600929730, 14346.114698914059563, 28502.272377756659505, 28835.636437354567322, 13473.368739266960527,2387.9894006598233823, 99.579680699439479278),
    (1.0, 38.702667724669835916, 579.20962089531281339, 4344.4743494790396875, 17740.385816044920402, 40220.102972229127618, 49665.114336529855022, 31552.009191297117961, 9222.0904372933369941, 992.44893809992759951,22.340799371096865866)
)

# pre-pack coefficients
const pack_EXPINT_ASYM_COEF = SIMDMath.pack_poly(EXPINT_ASYM_COEF)
const pack_EXPINT_POW_COEF = SIMDMath.pack_poly(EXPINT_POW_COEF)

# domain: x >= 0.0
function _expint(x::Float64)
    if x >= 1.0
        e = exp(-x/2)
        xinv = inv(x)
        p = SIMDMath.horner_simd(xinv, pack_EXPINT_ASYM_COEF)
        return (e * p[1] / p[2]) * e
    else
        p = SIMDMath.horner_simd(x, pack_EXPINT_POW_COEF)
        return (x * p[1] / p[2] - (Base.MathConstants.eulergamma + log(x)))
    end
end
function _expintx(x::Float64)
    if x >= 1.0
        xinv = inv(x)
        x > 1e14 && return xinv*evalpoly(xinv, (1.0, -1.0, 2.0, -6.0, 24.0))
        p = SIMDMath.horner_simd(xinv, pack_EXPINT_ASYM_COEF)
        return p[1] / p[2]
    else
        p = SIMDMath.horner_simd(x, pack_EXPINT_POW_COEF)
        return (x * p[1] / p[2] - (Base.MathConstants.eulergamma + log(x))) * exp(x)
    end
end
```

Reply:
This is about 15-20% faster. Again this was already auto-vectorizing well but we can increase the packing efficiency slightly. My big suggestion for these type of functions though would always to use the scaled versions then build up from there. So we should optimize and look at `besselkx` instead of `besselk` and then define `besselk` in terms of the scaled version. These function underflow fairly rapidly (like x > 750) so it's helpful to have the scaled versions rock solid especially since this is the asymptotic expansion `x >> nu`

Reply:
I wish I could get the minimax to work over the whole range of x and not need the straight asymptotic expansion for `x>1e14`. Mathematica giving me some finicky underflow errors can't seem to figure out.

Reply:
note that you don't want the non-packed constants in global scope since that will cause them to remain in memory.

Reply:
> note that you don't want the non-packed constants in global scope since that will cause them to remain in memory.

Ya probably just want to combine the evaluations so its like...
```julia
const pack_EXPINT_ASYM_COEF = SIMDMath.pack_poly(
    (1.3319248966682697359e-30, 0.99999999999999993242, 37.702667724670064050, 542.50695317051747114, 3836.6700640600929730, 14346.114698914059563, 28502.272377756659505, 28835.636437354567322, 13473.368739266960527,2387.9894006598233823, 99.579680699439479278),
    (1.0, 38.702667724669835916, 579.20962089531281339, 4344.4743494790396875, 17740.385816044920402, 40220.102972229127618, 49665.114336529855022, 31552.009191297117961, 9222.0904372933369941, 992.44893809992759951,22.340799371096865866)
)
```

Reply:
You're a machine. Thanks for hacking on this.